jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpClientTransportDynamic.java

```diff
@@ -117,11 +122,12 @@ public Origin newOrigin(Request request)
                 // Find the first protocol that matches the version.
                 for (String p : info.getProtocols(secure))
                 {
-                    if (wanted.contains(p))
-                    {
-                        matchingInfos.add(info);
+                    if (excludedProtocols.contains(p))
```

Comment:
Why is this case `break` and not `continue`?

Reply:
Because for an `info` with `protocols=[h2c-17,h2c]`, if the first is excluded, then also the second would be.